src/gridded/indexing.jl

```diff
@@ -49,6 +50,12 @@ end
     weightedindex(fs, degree(mode), knotvec, x, iclamp)
 end
 
+@inline function weightedindex_parts2(fs::F, mode::Gridded, knotvec::AbstractVector, x, i) where F
```

Comment:
Looks like this is essentially all of the previous method except the lookup of `i`. Rather than duplicate code, I'd rather see the previous one just call this function, i.e., previous method becomes

```julia
    i = find_knot_index(knotvec, x)
    return weightedindex_parts2(fs, mode, knotvec, x, i)
```